integration/integration_test.go

```diff
@@ -342,10 +387,40 @@ func TestBindsnoop(t *testing.T) {
 	t.Parallel()
 
 	bindsnoopCmd := &command{
-		name:           "StartBindsnoopGadget",
-		cmd:            fmt.Sprintf("$KUBECTL_GADGET trace bind -n %s", ns),
-		expectedRegexp: fmt.Sprintf(`%s\s+test-pod\s+test-pod\s+\d+\s+nc`, ns),
-		startAndStop:   true,
+		name:         "StartBindsnoopGadget",
+		cmd:          fmt.Sprintf("$KUBECTL_GADGET trace bind -n %s -o json", ns),
+		startAndStop: true,
+		customOutputVerifier: func(output string) error {
+			expected := bindTypes.Event{
+				Event:     buildBaseEvent(ns),
+				Comm:      "nc",
+				Protocol:  "TCP",
+				Addr:      "::",
+				Port:      9090,
+				Options:   ".R...",
+				Interface: "0",
+			}
+
+			return expectAtLeastOneMatch(output, expected, func(e bindTypes.Event) bool {
```

Comment:
Is `expectAtLeastOneMatch` really what we want for this test? It seems to me that all generated events should match. (We're running the command in a pod and filtering by namespace where there is a single pod, hence I wouldn't expect to capture any event that we didn't generate).

Reply:
Yeah, it is not but it was just the quickest way to implement a solution and start this discussion. The final solution would be to have multiple flavours. For instance:

- For execsnoop, we are triggering three different events and we would like to verify that every generated event is one of the expected ones. So, we would need something like `expectOneFrom`, which would receive a slice of expected events and every event must match one of them:

```bash
{"node":"worker","namespace":"test-execsnoop-2760712051242727130","pod":"test-pod","container":"test-pod","type":"normal","pid":245543,"ppid":245522,"mountnsid":4026532375,"pcomm":"sh","args":["/bin/sh","-c","while true; do date \u0026\u0026 sleep 0.1; done"]}
{"node":"worker","namespace":"test-execsnoop-2760712051242727130","pod":"test-pod","container":"test-pod","type":"normal","pid":245559,"ppid":245543,"mountnsid":4026532375,"pcomm":"date","args":["/bin/date"]}
{"node":"worker","namespace":"test-execsnoop-2760712051242727130","pod":"test-pod","container":"test-pod","type":"normal","pid":245560,"ppid":245543,"mountnsid":4026532375,"pcomm":"sleep","args":["/bin/sleep","0.1"]}
[... data and sleep event are repeated ...]
```
- For bindsnoop, they are all the same. So there we will need `expectOnly`, which would receive one expected event and every generated must match it:
```bash
{"node":"worker","namespace":"test-bindsnoop-3509468263802846636","pod":"test-pod","container":"test-pod","type":"normal","pid":239853,"comm":"nc","proto":"TCP","addr":"::","port":9090,"opts":".R...","if":"0","mountnsid":4026532375}
{"node":"worker","namespace":"test-bindsnoop-3509468263802846636","pod":"test-pod","container":"test-pod","type":"normal","pid":239881,"comm":"nc","proto":"TCP","addr":"::","port":9090,"opts":".R...","if":"0","mountnsid":4026532375}
[... All the following are equal ...]
```

WDYT?

Reply:
Makes sense.